spec/controllers/api/v1/labware_types_controller_spec.rb

```diff
@@ -12,11 +12,17 @@ def validate_labware_type(labware_type_json, labware_type)
   def validate_labware_included(layout_json, layout)
     expect(layout_json[:id]).to eq(layout.id.to_s)
     expect(layout_json[:attributes][:name]).to eq(layout.name)
+
+    expected_locations_json = layout_json[:relationships][:locations][:data]
+    expect(expected_locations_json.size).to eq(layout.locations.size)
+    expected_locations_json.zip(layout.locations).each do |json_location, orig_location|
+      expect(json_location[:id]).to eq(orig_location.id.to_s)
+    end
   end
 
   describe "GET #show" do
     it "should return a serialized labware_type instance" do
-      labware_type = create(:labware_type)
+      labware_type = create(:labware_type, layout: create(:layout_with_locations))
```

Comment:
If layout is mandatory it should be added to the factory. Otherwise the factory produces an invalid model.